hyperspy/_signals/eds.py

```diff
@@ -175,25 +177,61 @@ def sum(self, axis=None, out=None):
             return s
     sum.__doc__ = Signal1D.sum.__doc__
 
-    def rebin(self, new_shape, out=None):
-        new_shape_in_array = []
-        for axis in self.axes_manager._axes:
-            new_shape_in_array.append(
-                new_shape[axis.index_in_axes_manager])
-        factors = (np.array(self.data.shape) /
-                   np.array(new_shape_in_array))
-        s = super().rebin(new_shape, out=out)
-        s = out or s
-        # modify time per spectrum
-        this_md = self.metadata
-        that_md = s.metadata
-        keys = ("Acquisition_instrument.SEM.Detector.EDS.live_time",
-                "Acquisition_instrument.TEM.Detector.EDS.live_time")
-        for key in keys:
-            if key in this_md:
-                that_md.set_item(key, this_md.get_item(key) * np.prod(factors))
-        return s
-    sum.__doc__ = Signal1D.sum.__doc__
+    def rebin(self, new_shape=None, scale=None, crop=True, out=None):
+        """
+        %s
+        """
+
+    #Series of if statements to check that only one out of new_shape or scale
+    #has been given. New_shape is then converted to scale. If both or neither
+    #are given the function raises and error and wont run.
+
```

Comment:
Throughout the function, comparisons if something is `None` should be done using `is` operator, like so:
```python
if new_shape is None and scale is not None:
    # do stuff
```